### PR TITLE
feat: add SocialNetwork.Shared#utils.share to open share popup

### PR DESCRIPTION
**Pull request by UncleBill**
closes #MF-194

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Comment by UncleBill** [comment]
@DimensionDev/qa Please help to verify.

File: packages/mask/src/social-network-adaptor/utils.ts [hunk 1]
```diff
@@ -23,3 +23,18 @@ export const getCurrentIdentifier = () => {
         globalUIState.profiles.value[0]
     )
 }
+
+export const createCenterWindowConfig = (width: number, height: number) => {
+    const x = window.screenX + (window.innerWidth - width) / 2
+    const y = window.screenY + (window.innerHeight - height) / 2
+    return [
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/utils.ts:30]: 'noopener noreferrer'

File: packages/mask/src/social-network-adaptor/twitter.com/shared.ts [hunk 2]
```diff
@@ -33,6 +34,11 @@ export const twitterShared: SocialNetwork.Shared & SocialNetwork.Base = {
             },
         },
         getPostURL,
+        share(text) {
+            const config = createCenterWindowConfig(700, 520)
+            const url = this.getShareLinkURL!(text)
+            window.open(url, 'share', config) || window.location.assign(url)
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/shared.ts:32]:
```suggestion
            const url = this.getShareLinkURL?.(text)
            if (url) window.open(url, 'share', config) || window.location.assign(url)
```

File: packages/mask/src/social-network-adaptor/minds.com/shared.ts [hunk 3]
```diff
@@ -16,6 +16,10 @@ export const mindsShared: SocialNetwork.Shared & SocialNetwork.Base = {
         isValidUsername: usernameValidator,
         textPayloadPostProcessor: undefined,
         getPostURL,
+        share(message) {
+            const url = this.getShareLinkURL!(message)
+            window.open(url, '_blank', 'noopener noreferrer')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/minds.com/shared.ts:21]:
```suggestion
            const url = this.getShareLinkURL?.(message)
            if (url) window.open(url, '_blank', 'noopener noreferrer')
```

File: packages/mask/src/social-network-adaptor/facebook.com/shared.ts [hunk 4]
```diff
@@ -18,6 +18,10 @@ export const facebookShared: SocialNetwork.Shared & SocialNetwork.Base = {
         isValidUsername: (v) => !!isValidFacebookUsername(v),
         textPayloadPostProcessor: undefined,
         getPostURL,
+        share(message) {
+            const url = this.getShareLinkURL!(message)
+            window.open(url, '_blank', 'noopener noreferrer')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/shared.ts:23]:
```suggestion
            const url = this.getShareLinkURL?.(message)
            if (url) window.open(url, '_blank', 'noopener noreferrer')
```

File: packages/mask/src/social-network-adaptor/utils.ts (hunk 1 above)
Comment (UncleBill) [reply packages/mask/src/social-network-adaptor/utils.ts:30]: Popup will be blocked with these.

File: packages/mask/src/social-network-adaptor/facebook.com/shared.ts (hunk 4 above)
Comment (UncleBill) [reply packages/mask/src/social-network-adaptor/facebook.com/shared.ts:23]: `getShareLinkURL` is set below.

File: packages/mask/src/social-network-adaptor/minds.com/shared.ts (hunk 3 above)
Comment (UncleBill) [reply packages/mask/src/social-network-adaptor/minds.com/shared.ts:21]: `getShareLinkURL` is set below.

**Review by nuanyang233 — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Review by guanbinrui — APPROVED** [review]

**Comment by AinuX** [comment]
Checked

**Pull request merged**